miden-tx/src/lib.rs

```diff
@@ -40,9 +40,15 @@ pub use error::{
 
 pub mod auth;
 
+#[cfg(feature = "testing")]
+pub mod testing;
+
 #[cfg(test)]
 mod tests;
 
+#[cfg(test)]
+mod kernel_tests;
```

Comment:
Maybe not for this PR, but I think we can combine these into a single `tests` module (with `kernel` submodule).